specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/ManagedEnvironments.json

```diff
@@ -1828,6 +1828,11 @@
           "$ref": "./CommonDefinitions.json#/definitions/WorkloadProfileName",
           "description": "Workload profile type for the workloads to run on."
         },
+        "enableFips": {
```

Comment:
>enableFips [](http://example.com/codeflow?start=9&length=10)

ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean.

Reply:
`enableFips` is a property in request sent by customer to initiate fips mode. It does not represent the existing fips state. If the operation fails, the `provisioningState` will be set to failed and customer will have to set this property to false.

We are using the same practice done by other azure services. Refer
1. Application Gateway: https://github.com/Azure/azure-rest-api-specs/blob/8a287afb3721dee0d88f11502ec123470bc52a28/specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/applicationGateway.json#L1614-L1616
2. AKS: https://github.com/Azure/azure-rest-api-specs/blob/8a287afb3721dee0d88f11502ec123470bc52a28/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-04-01/managedClusters.json#L2864-L2866